### DISCUSSION TITLE
Mode-switcher button height

### DISCUSSION THREAD

**Discussion by s0la** [davatorium/rofi#1856] [Theming]
Hi folks,

I have a question about button height.. I tried to find the answer on my own, ofc, but came up short so I decided to try asking here..

Given the picture is worth a thousand words, I posted two images in order to better illustrate what I mean:

![rofi_mpd](https://github.com/davatorium/rofi/assets/6886981/a8639d21-3da0-4961-bf40-dd529939991e)

![rofi_ws](https://github.com/davatorium/rofi/assets/6886981/1e876277-fa97-47ee-b2cc-c278dd16688f)

So, by "mode-switcher button" I mean the buttons on the right side, vertically aligned.
As you can see on the first image, they are quite elongated, since there is only 2 buttons, opposed to second image where the buttons are much more square looking..

What I'd like to do is "enforce" that square size to be consistent regardless of the button number.. Is that even possible?

Thanks!

**Comment by DaveDavenport** [comment]
To not expand to available space, set `expand: false;` on the button.

**Reply by s0la** [reply]
Thanks for response, Dave!

I already tried that, but didn't get exactly what I wanted:

![expand](https://github.com/davatorium/rofi/assets/6886981/fe7598cf-1c1d-4bd7-aee1-bf96e00fb5f1)

It removes space around buttons, but buttons themselves are too tiny and not centered.. I tried adding height and padding, but size simply always stays the same :/

Is there a way to set fixed height when buttons are not expanded, and can I align they vertically to position them in the center of mode-switcher?

EDIT:

I managed to set the padding to the button, the issue previously was that I didn't have `button` element on its own, I just had `button normal` and `button selected`, so that part is solved, now it looks like this:

![padding](https://github.com/davatorium/rofi/assets/6886981/9238fb56-3156-4d43-ae6e-bc6e30323122)

The only remaining part would be to center them vertically, if that's possible :)

**Reply by DaveDavenport** [reply]
https://davatorium.github.io/rofi/1.7.5/rofi-theme.5/#advanced-box-packing

**Reply by s0la** [reply]
Excellent, that did the trick, works perfectly!
Thank you so much for your time, no way I would be able to figure this out without your help!
And thanks for your work in general and for bringing new awesome features, can't imagine a day without rofi :)
